File: src/vb/Samples/ApiSamples/SampleProxy.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Threading

Namespace ApiSamples
	Friend Class SampleProxy
		Inherits MarshalByRefObject

		Public Sub RunSample(ByVal method As MethodInfo, ByVal parameters() As Object, ByVal consoleOut As TextWriter)
			If consoleOut IsNot Nothing Then
				Console.SetOut(consoleOut)
			End If

			method.Invoke(Nothing, parameters)
		End Sub
	End Class
End Namespace

File: cs/SolidEdge/ApiDemos/ApiDemos/SampleProxy.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading;

namespace ApiDemos
{
    class SampleProxy : MarshalByRefObject
    {
        public void RunSample(MethodInfo method, object[] parameters, TextWriter consoleOut)
        {
            if (consoleOut != null)
            {
                Console.SetOut(consoleOut);
            }

            method.Invoke(null, parameters);
        }
    }
}
